scripts/tb-bandwidth.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+#
+# dp_tunnel_status() - Reads DisplayPort tunnel status
+# @domain: Domain number
+# @router: Route string of the router
+# @adapter: DisplayPort IN adapter
+#
+dp_tunnel_status() {
+    local domain=$1
+    local route=$2
+    local adapter=$3
+    local val
+
+    # Check type first
+    val=$(tbget -d $domain -r $route -a $adapter ADP_CS_2)
+    val=$(printf "0x%x" $((val & 0xffffff)))
+    case $val in
+    0xe0101 | 0xe0102)
+        ;;
+    *)
+        echo "Error: unsupported adapter type: $val" 1>&2
+        exit 1
+        ;;
+    esac
+
+    # Check if paths are enabled
+    val=$(tbget -d $domain -r $route -a $adapter ADP_DP_CS_0.VE)
+    if [ $val != "0x1" ]; then
+        printf "\t\tTunnel    : inactive\n"
+        return
+    fi
+
+    # read link rate and line count
+    read -r lanes lrate <<EOF
+$(tbget -D -d $domain -r $route -a $adapter "DP_STATUS.Lane Count" \
+                         "DP_STATUS.Link Rate" |  \
+                        xargs)
+EOF
+    case $lrate in
+        0)
+            lrate=1620
+            hbr="RBR"
+            ;;
+        1)
+            lrate=2700
+            hbr="HBR"
+            ;;
+        2)
+            lrate=5400
+            hbr="HBR2"
+            ;;
+        3)
+            lrate=8100
+            hbr="HBR3"
+            ;;
+        *)
+            echo "Error: unsupported rate: $lrate" 1>&2
+            exit 1
+            ;;
+    esac
+
+    # Is BW alloc mode enabled?
+    val=$(tbget -d $domain -r $route -a $adapter ADP_DP_CS_8.DPME)
+    if [ $val = "0x1" ]; then
+        read -r granularity  allocated <<EOF
+$(tbget -d $domain -r $route -a $adapter "ADP_DP_CS_2.GR" \
+                                         "DP_STATUS.Allocated BW" |
+                                         xargs)
+EOF
+        case $granularity in
+            0x0)
+                granularity=250
+                ;;
+            0x1)
+                granularity=500
+                ;;
+            0x2)
+                granularity=1000
+                ;;
+            *)
+                echo "Error: unsupported granularity: $granularity" 1>&2
+                exit 1
+                ;;
+        esac
+        printf "\t\tTunnel    : active\n"
+        printf "\t\tDPBWMode  : enabled\n"
+        rate=$((granularity * allocated))
+        rate_gbps=$(bc <<< "scale=2; $rate / 1000")
```

Comment:
You can use shell arithmetic here instead of bc I think like $((rate / 1000)) and so on.

Reply:
I have made new changes and uploaded changes. please review new changes